rust/lance-io/src/object_writer.rs

```diff
@@ -152,6 +154,10 @@ impl UploadState {
             _ => unreachable!(),
         };
     }
+
+    fn any_to_abort(&mut self) -> Self {
```

Comment:
Hi, I have a feeling that we don't need to have this?

Reply:
I found this ```impl UploadState``` block is used for '/// Methods for state transitions'. So I think we can put all state transitions together, including 'from any state to abort', for better readability.

Maybe the method name 'any_to_abort' is not clear? Would 'force_to_abort' be better?

 What do you think?